### PR TITLE
Accelerate pixmaps bigger than 4k pixels.

### PR DESCRIPTION
**Pull request by xjuan**
This is needed for EGL_KHR_image_pixmap to work on pixmaps other than
scanout and windows.

https://phabricator.endlessm.com/T15893

### CODE REVIEW

File: src/armsoc_exa.c [hunk 1]
```diff
@@ -178,7 +181,7 @@ ARMSOCCreatePixmap2(ScreenPtr pScreen, int width, int height,
 	}
 	priv->usage_hint = usage_hint;
 
-	if (is_accel_pixmap(priv))
+	if (is_accel_pixmap(priv, width * height))
```
Comment (magcius) [inline src/armsoc_exa.c]: A more accurate version would be `new_fb_pitch * height * bitsPerPixel / 8` to represent the malloc size.

**Comment by dsd** [comment]
Accelerated pixmaps will come from CMA which is a precious resource. Before merging this I'd like to measure the impact i.e. see how many more pixmaps we are accelerating that we previously were not.

**Comment by magcius** [comment]
I thought we had non-contiguous pixmap allocation working?

File: src/armsoc_exa.c (hunk 1 above)
Comment (magcius) [inline src/armsoc_exa.c]: Well the benchmarks you did were on allocation speed, right? So I would think the decision should be based on size.

**Comment by dsd** [comment]
Good point.. perhaps not such a big deal then, just double check that we aren't doing that from CMA

**Pull request merged**